### ISSUE TITLE
Moleculenet dataset fails to featurize--Sine_Coulomb_Matrix runtime error

### ISSUE THREAD

**Issue by alanatransrights** [deepchem/deepchem#2162]
VERSION: Deepchem Nightly
ENVIRONMENT: Google Colab

I am attempting to load a dataset from MoleculeNet, but there seems to be a problem with featurization.

My script so far, in Google Colab, based on the MoleculeNet documentation:

```
#Env Setup
!pip install scipy
!pip install imgaug
!pip install pymatgen
!pip install --pre deepchem
!pip install tensorflow

import deepchem as dc
tasks, datasets, transformers = dc.molnet.load_perovskite(reload=False)
train_dataset, val_dataset, test_dataset = datasets
n_tasks = len(tasks)
n_features = train_dataset.get_data_shape()[0]
model = dc.models.MultitaskRegressor(n_tasks, n_features)
```

When ran, the output reads:

```
Failed to featurize datapoint 0. Appending empty array
Failed to featurize datapoint 1. Appending empty array
Failed to featurize datapoint 2. Appending empty array
Failed to featurize datapoint 3. Appending empty array
...
```

I have reset my environment multiple times. Do I need to specify a featurizer in the call to dc.molnet, perhaps?

**Comment by nd-02110114** [comment]
Could you try to install `matminer`? The default featurizer is `SineCoulombMatrix`, and it needs `matiminer`

```bash
!pip install matminer
```

**Comment by alanatransrights** [comment]
Thank you! That certainly solves part of the problem. Now, it casts a ComplexError:

```
/usr/local/lib/python3.6/dist-packages/deepchem/feat/material_featurizers/sine_coulomb_matrix.py:90: ComplexWarning:

Casting complex values to real discards the imaginary part

/usr/local/lib/python3.6/dist-packages/deepchem/trans/transformers.py:538: RuntimeWarning:

invalid value encountered in true_divide
```

Which seems to still be a problem with `SinceCoulombMatrix`.

**Comment by nd-02110114** [comment]
I think `ComplexWarning` is not serious problem. 

This msg shows some inorganic crystal structures lead to the coulomb_matrix contains complex number. 
Such a situation sometime happens, so this is not `Error`, just `Warning`.

**Comment by alanatransrights** [comment]
Hmm... I see. 

Forgive my ignorance: is there some quick test I could run to see if my dataset is successfully loaded such that I can know if this issue is resolved?

**Comment by nd-02110114** [comment]
Basically, I think the data loading is successfully if you don't face any runtime errors. (not warning)
You can check dataset values like this.


```python
print(train_dataset.X[0])
print(train_dataset.X[1])
```

In addtion to this, If you want to check whether the featurized values are correct or not, I think it is better to construct model with them and check the accuracy.

**Comment by alanatransrights** [comment]
Thank you so much for the help! 

The output for those two lines are:

```
[ 1.33514126 -0.69242071  5.65577459 25.49495848  0.93306779  0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.        ]
[ 1.41808184 -0.40013617  2.76953358  1.00736876  0.81146693  0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.          0.          0.
  0.          0.          0.          0.        ]
```

That looks to strange to me! A lot of zeroes... Is it normal?

**Comment by nd-02110114** [comment]
This is also not problem. 0 is pad value to ensure that all feature vectors are equal length.

Please see https://deepchem.readthedocs.io/en/latest/featurizers.html#deepchem.feat.SineCoulombMatrix

And, I seem you are newcomer for DeepChem and Machine Learning. So, firstly, I recommend you try DeepChem tutorials. You can understand how to use DeepChem APIs more deeply.
https://github.com/deepchem/deepchem/tree/master/examples/tutorials


In addition to this, if you are interested in inorganic materials, I think matminer examples are also useful
https://nbviewer.jupyter.org/github/hackingmaterials/matminer_examples/blob/master/matminer_examples/index.ipynb

**Comment by alanatransrights** [comment]
Thank you so much! You have been super helpful!! 

I will be sure to read up and go through the resources you have provided. 

Have a wonderful day!
